File: production/client/src/main/scala/kpn/client/components/changes/ChangesState.scala
// TODO migrate to Angular
package kpn.client.components.changes

import kpn.client.components.common.PageState
import kpn.shared.ChangesPage
import kpn.shared.changes.filter.ChangesParameters

case class ChangesState(
  pageState: PageState[ChangesPage] = PageState(),
  parameters: Option[ChangesParameters] = None
)

File: frontend/src/app/analysis/changes/store/changes.state.ts
import { ChangesPage } from '@api/common';
import { ChangesParameters } from '@api/common/changes/filter';
import { ApiResponse } from '@api/custom';

export const initialState: ChangesState = {
  changesParameters: null,
  changesPage: null,
};

export interface ChangesState {
  changesParameters: ChangesParameters;
  changesPage: ApiResponse<ChangesPage>;
}

export const changesFeatureKey = 'changes';
